robotiq_c_model_control/src/robotiq_c_model_control/robotiq_gripper_action_interface.py

```diff
@@ -498,6 +497,44 @@ def send_gripper_command(self, goal, parse=True):
             rospy.signal_shutdown('Error While Sending Command - Shutting Down')
             return False
 
+    def shutdown_grip_state_service(self):
+        with self.grip_state_lock_:
+            if self.grip_state_srv_ is not None:
+                self.grip_state_srv_.shutdown()
+                self.grip_state_srv_ = None
+
+    def handle_grip_state_service(self, request):
+        with self.grip_state_lock_:
+            resp = GetRobotiqGripStateResponse()
+            if self.resetting:
+                resp.state = resp.UNKNOWN
+            else:
+                registers = self.wait_for_next_status()
+                moved = registers.gGTO != 0
+                if moved and registers.gOBJ == 1:
+                    # Fingers stopped due to contact while opening
+                    resp.state = GetRobotiqGripStateResponse.OPEN
+                elif moved and registers.gOBJ == 2:
+                    # Fingers stopped due to contact while closing
+                    resp.state = GetRobotiqGripStateResponse.CLOSED
+                elif moved and registers.gOBJ == 3:
+                    # Fingers moved to the requested position without detecting an
+                    # object. Use the gripper position to determine the current
+                    # grip state.
+                    position = int(registers.gPO)
+                    position_median = int(abs(self.max_closed - self.max_open) / 2.0)
```

Comment:
I would add back in the calibration offsets here as that will shift the median.

Reply:
Just so I'm clear, you mean:
`position_median = int(abs(self.max_closed - self.max_open) / 2.0) + self.max_open`

Reply:
or did you mean:
`position_median = int(abs(self.max_closed - self.max_open) / 2.0) + (255 - (self.max_closed - self.max_open)`

Reply:
Sorry. I confused myself. What you had originally is fine. I forgot that max_open and max_closed are updated on calibration. That confusion is probably due to a poor naming choice on my part when it was initially written.

Reply:
OK! I was scratching my head trying to come up with something :)